test/integration/discovery.go

```diff
@@ -306,32 +309,165 @@ var _ = Describe("Discover Cluster", func() {
 			entityDTOs, _, err := discoveryClient.DiscoverWithNewFramework("discovery-integration-test")
 			framework.ExpectNoError(err, "Failed completing discovery of test cluster")
 
+			podEntityDTO, nodeEntityDTO = getPodNodeDTO(entityDTOs, podFullName)
+
 			//Validation logic
-			if !validateBuyerSellerCommodity(entityDTOs, nodeNameForPod, podFullName, proto.CommodityDTO_VMPM_ACCESS, "foo in (bar)") {
+			if !validateBuyerSellerCommodity(podEntityDTO, nodeEntityDTO, nodeNameForPod, proto.CommodityDTO_VMPM_ACCESS, "foo in (bar)") {
 				framework.Failf("PV affinity is not honored")
 			}
 		})
 	})
 
+	Describe("discovering pod with pv having node with zone label", func() {
```

Comment:
What would be your thoughts on merging this test with the previous one (on line 273).
I am seeing continuous increase in the steps that we are doing in the integration test thus linearly increasing the overall time it takes to complete the test.

Here we are almost all steps same as previous test, except the node has a zone label now, but creating the test workload twice (which takes considerable amount of time).

Reply:
@irfanurrehman I agree. Somewhere I had thought this but didn't implement thinking we might want to test pv with affinity exclusively. I have updated the code to create a single test setup for all 3 scenarios. Please review. Thanks for your feedback